app/models/authentication.rb

```diff
@@ -81,6 +81,18 @@ def raise_event(status, _message = nil)
     MiqEvent.raise_evm_event_queue(ci, "#{prefix}_auth_#{status}")
   end
 
+  def assign_values(options)
```

Comment:
An additional option is to have dedicated setters at the subclass level, and then `attributes=` would just call those setters.  This way you also don't need the column verification.

@jrafanie Thoughts here?